diff --git a/network-manager-1.30.6/debian/rules b/network-manager-1.30.6/debian/rules
index 55e2e50..114c33a 100755
--- a/network-manager-1.30.6/debian/rules
+++ b/network-manager-1.30.6/debian/rules
@@ -65,8 +65,11 @@ override_dh_install:
 override_dh_missing:
 	dh_missing --fail-missing
 
+override_dh_shlibdeps:
+	dh_shlibdeps --dpkg-shlibdeps-params=--ignore-missing-info
+
 override_dh_makeshlibs:
-	dh_makeshlibs -X/usr/lib/$(DEB_HOST_MULTIARCH)/NetworkManager/ -X/usr/lib/pppd/
+	dh_makeshlibs -X/usr/lib/$(DEB_HOST_MULTIARCH)/NetworkManager/ -X/usr/lib/pppd/ 
 
 override_dh_installsystemd:
 	dh_installsystemd -pnetwork-manager --no-start NetworkManager-dispatcher.service NetworkManager-wait-online.service
