diff --git a/src/core/devices/wifi/nm-device-wifi.c b/src/core/devices/wifi/nm-device-wifi.c
index 4377283458..15803aebfc 100644
--- a/src/core/devices/wifi/nm-device-wifi.c
+++ b/src/core/devices/wifi/nm-device-wifi.c
@@ -2232,15 +2232,14 @@ wifi_secrets_get_secrets(NMDeviceWifi                *self,
 {
     NMDeviceWifiPrivate *priv = NM_DEVICE_WIFI_GET_PRIVATE(self);
     NMActRequest        *req;
+      wifi_secrets_cancel(self);
 
-    wifi_secrets_cancel(self);
-
-    req = nm_device_get_act_request(NM_DEVICE(self));
-    g_return_if_fail(NM_IS_ACT_REQUEST(req));
+      req = nm_device_get_act_request(NM_DEVICE(self));
+      g_return_if_fail(NM_IS_ACT_REQUEST(req));
 
-    priv->wifi_secrets_id =
+      priv->wifi_secrets_id =
         nm_act_request_get_secrets(req, TRUE, setting_name, flags, NULL, wifi_secrets_cb, self);
-    g_return_if_fail(priv->wifi_secrets_id);
+      g_return_if_fail(priv->wifi_secrets_id);
 }
 
 /*
@@ -2347,6 +2346,11 @@ need_new_wpa_psk(NMDeviceWifi              *self,
 
     g_return_val_if_fail(connection, FALSE);
 
+    _LOGI(LOGD_DEVICE | LOGD_WIFI,
+              "Activation: (wifi) disconnected during association (%d)", disconnect_reason);
+    _LOGI(LOGD_DEVICE | LOGD_WIFI,
+              "Activation: (wifi) disconnected during association, not disregarding PSK - Athom fix");
+
     /* A bad PSK will cause the supplicant to disconnect during the 4-way handshake */
     if (old_state != NM_SUPPLICANT_INTERFACE_STATE_4WAY_HANDSHAKE)
         return FALSE;
@@ -2355,7 +2359,7 @@ need_new_wpa_psk(NMDeviceWifi              *self,
     if (s_wsec)
         key_mgmt = nm_setting_wireless_security_get_key_mgmt(s_wsec);
 
-    if (g_strcmp0(key_mgmt, "wpa-psk") == 0) {
+    if (g_strcmp0(key_mgmt, "wpa-psk") == 0 && false) {
 /* -4 (locally-generated WLAN_REASON_DISASSOC_DUE_TO_INACTIVITY) usually
          * means the driver missed beacons from the AP.  This usually happens
          * due to driver bugs or faulty power-save management.  It doesn't
@@ -2364,6 +2368,9 @@ need_new_wpa_psk(NMDeviceWifi              *self,
 #define LOCAL_WLAN_REASON_DISASSOC_DUE_TO_INACTIVITY -4
         if (disconnect_reason == LOCAL_WLAN_REASON_DISASSOC_DUE_TO_INACTIVITY)
             return FALSE;
+#define LOCAL_WLAN_REASON_4WAY_HANDSHAKE_TIMEOUT -15
+        if (disconnect_reason == LOCAL_WLAN_REASON_4WAY_HANDSHAKE_TIMEOUT)
+            return FALSE;
 
         *setting_name = NM_SETTING_WIRELESS_SECURITY_SETTING_NAME;
         return TRUE;
